src/python/deepgnn/graph_engine/graph_ops.py

```diff
@@ -177,3 +177,18 @@ def sub_graph(
             adj += np.eye(n)
 
         return unique_nodes, adj, src_nodes_idx
+
+
+def edge_sub_graph(graph: Graph, edges: np.ndarray, num_neighbors: List[int]):
+    """Sub graph of edges sampled from starting edges."""
+    edges = np.array(edges)
+    subgraph = edges
+    for n in num_neighbors:
+        _, dst, types = edges[:, 0], edges[:, 1], edges[:, 2]
```

Comment:
Push this line outside of loop? I don't see how any of `dst`, `types`, `edges` is updated inside loop.

Reply:
Fixed, now edges is updated every loop.